File: services/api/src/atlas_api/models/enums.py
"""Domain enumerations — mirrors packages/shared/src/types/enums.ts."""

from enum import StrEnum


class SourceKind(StrEnum):
    ARTICLE = "article"
    PDF = "pdf"
    REPOSITORY = "repository"
    IMAGE_SET = "image_set"
    DATASET = "dataset"
    TRANSCRIPT = "transcript"
    WEB_CLIP = "web_clip"


class SourceStatus(StrEnum):
    PENDING = "pending"
    INGESTING = "ingesting"
    READY = "ready"
    FAILED = "failed"


class VaultNoteKind(StrEnum):
    SOURCE = "source"
    ENTITY = "entity"
    CONCEPT = "concept"
    INDEX = "index"
    TIMELINE = "timeline"


class JobKind(StrEnum):
    INGEST = "ingest"
    COMPILE = "compile"
    INDEX = "index"
    ANSWER = "answer"
    EXPORT = "export"
    PUBLISH = "publish"
    HEALTH_CHECK = "health_check"
    SIMULATION = "simulation"


class JobStatus(StrEnum):
    QUEUED = "queued"
    RUNNING = "running"
    SUCCEEDED = "succeeded"
    FAILED = "failed"
    CANCELLED = "cancelled"


class OutputArtifactKind(StrEnum):
    BRIEF = "brief"
    SLIDE_DECK = "slide_deck"
    CHART = "chart"
    PUBLISHED_PAGE = "published_page"
    SIMULATION_PACKAGE = "simulation_package"
    DOWNLOAD = "download"


class WorkspaceRole(StrEnum):
    OWNER = "owner"
    ADMIN = "admin"
    EDITOR = "editor"
    REVIEWER = "reviewer"
    OBSERVER = "observer"


class EntityKind(StrEnum):
    PERSON = "person"
    COMPANY = "company"
    PROJECT = "project"
    PAPER = "paper"
    DATASET = "dataset"
    PRODUCT = "product"

File: packages/shared/src/types/enums.ts
export const SourceKind = {
  Article: "article",
  Pdf: "pdf",
  Repository: "repository",
  ImageSet: "image_set",
  Dataset: "dataset",
  Transcript: "transcript",
  WebClip: "web_clip",
  Audio: "audio",
  Image: "image",
  Video: "video",
  Docx: "docx",
  Xlsx: "xlsx",
  Pptx: "pptx",
} as const;
export type SourceKind = (typeof SourceKind)[keyof typeof SourceKind];

export const SourceStatus = {
  Pending: "pending",
  Ingesting: "ingesting",
  Ready: "ready",
  Failed: "failed",
} as const;
export type SourceStatus = (typeof SourceStatus)[keyof typeof SourceStatus];

export const VaultNoteKind = {
  Source: "source",
  Entity: "entity",
  Concept: "concept",
  Index: "index",
  Timeline: "timeline",
  Output: "output",
} as const;
export type VaultNoteKind = (typeof VaultNoteKind)[keyof typeof VaultNoteKind];

export const JobKind = {
  Ingest: "ingest",
  Compile: "compile",
  Index: "index",
  Answer: "answer",
  Export: "export",
  Publish: "publish",
  HealthCheck: "health_check",
  Simulation: "simulation",
  GenerateOutput: "generate_output",
  Research: "research",
} as const;
export type JobKind = (typeof JobKind)[keyof typeof JobKind];

export const JobStatus = {
  Queued: "queued",
  Running: "running",
  Succeeded: "succeeded",
  Failed: "failed",
  Cancelled: "cancelled",
} as const;
export type JobStatus = (typeof JobStatus)[keyof typeof JobStatus];

export const OutputArtifactKind = {
  Brief: "brief",
  SlideDeck: "slide_deck",
  Chart: "chart",
  PublishedPage: "published_page",
  SimulationPackage: "simulation_package",
  Download: "download",
  StatusReport: "status_report",
  ClientBrief: "client_brief",
  WeeklyDigest: "weekly_digest",
  RiskRegister: "risk_register",
  RaciMatrix: "raci_matrix",
  FollowupEmail: "followup_email",
  MermaidDiagram: "mermaid_diagram",
  Custom: "custom",
} as const;
export type OutputArtifactKind =
  (typeof OutputArtifactKind)[keyof typeof OutputArtifactKind];

export const WorkspaceRole = {
  Owner: "owner",
  Admin: "admin",
  Editor: "editor",
  Reviewer: "reviewer",
  Observer: "observer",
} as const;
export type WorkspaceRole = (typeof WorkspaceRole)[keyof typeof WorkspaceRole];
